src/main/java/org/folio/ed/service/RemoteStorageService.java

```diff
@@ -21,18 +25,29 @@
 @RequiredArgsConstructor
 public class RemoteStorageService {
   private static final String STAGING_DIRECTOR_NAME = "Dematic_SD";
+  private static final String STORAGE_ID = "storageId=";
+
+  private final Map<String, List<RetrievalQueueRecord>> retrievalsMap = new HashMap<>();
 
   private final RemoteStorageClient remoteStorageClient;
   private final AccessionQueueRecordToAsrItemConverter accessionQueueRecordToAsrItemConverter;
   private final RetrievalQueueRecordToAsrRequestConverter retrievalQueueRecordToAsrRequestConverter;
 
   public List<AccessionQueueRecord> getAccessionQueueRecords(String storageId) {
-    return remoteStorageClient.getAccessionsByQuery("storageId=" + storageId + "&accessioned=false")
+    return remoteStorageClient.getAccessionsByQuery(STORAGE_ID + storageId + "&accessioned=false")
       .getResult();
   }
 
-  public void setAccessionedByBarcode(String barcode) {
-    remoteStorageClient.setAccessionedByBarcode(barcode);
+  public List<RetrievalQueueRecord> getRetrievalQueueRecords(String storageId) {
+    retrievalsMap.put(storageId, remoteStorageClient.getRetrievalsByQuery(STORAGE_ID + storageId + "&retrieved=false")
+      .getResult());
+    return retrievalsMap.get(storageId);
```

Comment:
In case retrievalsMap will have  storageId key should not it return that result without additional execution of line 42 ?

Reply:
Good catch, thanks!